crates/ruma-events/src/room/join_rules.rs

```diff
@@ -64,7 +64,8 @@ pub enum JoinRule {
     #[serde(rename = "invite")]
     Invite,
 
-    /// Reserved but not yet implemented by the Matrix specification.
+    /// Users can join the room if they are invited, or they can request an invite to the room.
+    /// They can be allowed (invited) or denied (kicked/banned) access.
```

Comment:
So after sbd. knocks and is allowed in, they're in invited state and need to confirm once again themselves to join the room?

Also, please separate this into a new paragraph:
```suggestion
    ///
    /// They can be allowed (invited) or denied (kicked/banned) access.
```

Reply:
> So after sbd. knocks and is allowed in, they're in invited state and need to confirm once again themselves to join the room?

The spec says this:

> Homeservers are permitted to automatically accept invites as a result of knocks as they should be aware of the user’s intent to join the room. If the homeserver is not auto-accepting invites (or there was an unrecoverable problem with accepting it), the invite is expected to be passed down normally to the client to handle. Clients can expect to see the join event if the server chose to auto-accept.

Reply:
So the client will effectively receive either an invite or joined membership event.

Reply:
Maybe because of federation complications (only the user's own homeserver can change the user's state to joined)? Anyways, wording is fine then.